### PR TITLE
Add loading of ca certificates to plugin

### PR DESCRIPTION
**Pull request by nac2791**
Using internal server which requires certificates to authenticate. Tested internally and works just fine.

cc @jbrockopp @jmccann

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -127,6 +127,16 @@ func main() {
 			Usage:  "s3 secret key",
 			EnvVar: "PLUGIN_SECRET_KEY,CACHE_S3_SECRET_KEY,AWS_SECRET_ACCESS_KEY",
 		},
+		cli.StringFlag{
+			Name:   "ca_cert",
+			Usage:  "ca cert to connect to s3 server",
+			EnvVar: "PLUGIN_CA_CERT,CACHE_S3_CA_CERT,AWS_CA_CERT",
```
Comment (jmccann) [inline main.go]: Do we need/want `AWS` prefixed variables?

File: main.go [hunk 2]
```diff
@@ -127,6 +127,16 @@ func main() {
...
+		},
+		cli.StringFlag{
+			Name:   "ca_cert_path",
+			Usage:  "ca cert to connect to s3 server",
+			EnvVar: "PLUGIN_CA_CERT_PATH,CACHE_S3_CA_CERT_PATH,AWS_CA_CERT_PATH",
```
Comment (jmccann) [inline main.go]: Do we need/want AWS prefixed variables?

Comment (donny-dont) [reply main.go]: No

**Comment by donny-dont** [comment]
@jbrockopp @jmccann I don't think those are standard variables that users of AWS expect like they would with `AWS_ACCESS_KEY_ID` so lets not make more environment variables.

**Comment by jbrockopp** [comment]
@donny-dont @jmccann that sounds good. We weren't sure so we figured we would just add them initially and remove them later if its deemed a problem. We'll make a commit to remove those and ping you after its done.

**Comment by nac2791** [comment]
@jmccann @donny-dont Updated

**Comment by tboerger** [comment]
Instead of doing it like that or should be much better to include your custom ca certificates into all containers... Drone got an env variable to add custom volumes to all containers... That's how I'm doing that. Otherwise we end up with adding these options to various plugins...

**Comment by bradrydzewski** [comment]
yep, I would instead recommend using the global volume feature to mount certs into containers.

**Comment by jmccann** [comment]
So then the `/path/on/host` file specified in `DRONE_VOLUME` on the `server` would actually need to exist on all the `agent`s, right?

**Comment by tboerger** [comment]
That's right, it doesn't need to exist on the server, only on the agents. But if you are using a custom root certificate somebody can expect that it is anyway present on the host.

**Comment by donny-dont** [comment]
@jmccann can we remove this code now?

**Comment by jmccann** [comment]
@donny-dont To be honest I'm not sure.  @jbrockopp would be better suited to answer that now.

**Comment by tboerger** [comment]
IMHO this had never been added to the plugin. This is not a plugin logic...

**Comment by jbrockopp** [comment]
@donny-dont @tboerger you can remove the logic if you feel it doesn't belong.

@bradrydzewski is their anyway to "turn off" the `DRONE_VOLUME` for a specific step?

The scenario I'm concerned with is customers needing to inject their own CA cert for a specific step and being unable to since the volume would be mounted in.

**Comment by jbrockopp** [comment]
@tboerger @donny-dont do either of you have thoughts on the above?

> ... is their anyway to "turn off" the DRONE_VOLUME for a specific step?

The scenario I'm concerned with is customers needing to inject their own CA cert for a specific step and being unable to since the volume would be mounted in.

**Comment by jbrockopp** [comment]
@bradrydzewski @tboerger @donny-dont ping to see if you have thoughts on ^^. 

I understand the wanting to remove this feature but if there isn't a way to "turn off" the `DRONE_VOLUME` step then I would argue that they don't provide the same functionality and this feature should remain in the plugin.

Thoughts?

**Pull request merged**